include/freertos/stropts.h

```diff
@@ -38,12 +38,14 @@
 extern "C" {
 #endif
 
+#ifndef ESP32
```

Comment:
#ifndef Arduino, ...or something?  Or possibly the right action is to remove this file dependency for non POSIX usage.

Reply:
Ideally yes it would be good to drop the dependency. At this time though, it compiles with this declaration dropped as it is already declared in the ESP-IDF framework with a different datatype for key (int vs unsigned long int here). I'm also not 100% convinced that we can't use the POSIX approach on the ESP32 as the VFS layer supports it as does LwIP. That will only leave the HW CAN controller which has its own API which may be possible to wrap but I don't think that is entirely necessary at this stage.